scripts/busybox_test.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/python3
+import subprocess
+import os
+from helper_functions import *
+from test_api_calls import *
+import time
+
+
+ALCOR_SCRIPTS = os.path.dirname(os.path.abspath(__file__))
+os.chdir("..")
+ALCOR_ROOT = os.getcwd()
+ALCOR_SERVICES = ALCOR_ROOT +"/services"
+services_list = ['ignite', 'vpm', 'snm', 'rm', 'pim' ,'mm' ,'pm' ,'nm', 'sgm', 'ag', 'dpm', 'eim', 'qm', 'nam','gm']
+ser_port={}
+ip_mac={}
+
+def build_containers():
+  container_list =[]
+  print("building container images")
+  mvn_build = "mvn -Dmaven.test.skip=true -DskipTests clean package install"
+  docker_build_ignite = "docker build -t ignite-11 -f" + ALCOR_ROOT + "/lib/ignite.Dockerfile " +  ALCOR_ROOT + "/lib"
+  # 9001 vpm
+  docker_build_vpm ="docker build -t vpm " + ALCOR_SERVICES + "/vpc_manager/"
+  #9002 snm
+  docker_build_snm = "docker build -t snm " + ALCOR_SERVICES + "/subnet_manager/"
+  #9003 rm
+  docker_build_rm = "docker build -t rm " + ALCOR_SERVICES + "/route_manager/"
+  # 9004 pim
+  docker_build_pim = "docker build -t pim " + ALCOR_SERVICES + "/private_ip_manager/"
+  #9005 mm
+  docker_build_mm ="docker build -t mm " +  ALCOR_SERVICES + "/mac_manager/"
+  #9006 pm
+  docker_build_pm = "docker build -t pm " + ALCOR_SERVICES + "/port_manager/"
+  #9007 nm
+  docker_build_nm = "docker build -t nm " + ALCOR_SERVICES + "/node_manager/"
+  #9008 sgm
+  docker_build_sgm = "docker build -t sgm " + ALCOR_SERVICES + "/security_group_manager/"
+  #9009 ag
+  docker_build_ag = "docker build -t ag " + ALCOR_SERVICES + "/api_gateway/"
+  #9010 dpm
+  docker_build_dpm = "docker build -t dpm " + ALCOR_SERVICES + "/data_plane_manager/"
+  #9011 eip
+  docker_build_eip = "docker build -t eip " + ALCOR_SERVICES + "/elastic_ip_manager/"
+  #9012 qm
+  docker_build_qm = "docker build -t qm " + ALCOR_SERVICES + "/quota_manager/"
+  #9013 nam
+  docker_build_nam = "docker build -t nam " + ALCOR_SERVICES + "/network_acl_manager/"
+  #9014 nam
+  docker_build_ncm = "docker build -t ncm " + ALCOR_SERVICES + "/network_config_manager/"
+  #9015 gm
+  docker_build_gm = "docker build -t gm " + ALCOR_SERVICES + "/gateway_manager/"
```

Comment:
Could we have configuration file which this script could read, load and then build/run services? This way, it could be easier to add more services without changing the busybox_test.py

The configuration file could include
{service short name}, {port number}, {service directory name} etc.

Reply:
An example: 

https://github.com/futurewei-cloud/alcor/blob/3083c16a0f7fe0abf21fe305adf193f5025c0b4d/scripts/build.sh#L45